content/articles/creating-a-pyspark-data-frame-a-beginners-guide/index.md

```diff
@@ -0,0 +1,159 @@
+ourMarks_df = sparkSession.createDataFrame(resilientDistributedDataset, schema=subjects)
+```
+In this case, the RDD is sent in the form of data. The schema parameter of the `createDataFrame()` method will be filled with previously produced string topics.
+
+We'll look at the data type of `ourMarks_df` below to see if everything went well. 
+```Python
+print(type(ourMarks_df))
+```
+Output:
+```bash
+<class 'pyspark.sql.dataframe.DataFrame'>
+```
+Using the `printSchema()` method, we can analyze our schema, which is useful when we have many columns.
+```Python
+ourMarks_df.printSchema()
+```
+Output:
+```bash
+root
+ |-- Division: string (nullable = true)
+ |-- Kiswahili: long (nullable = true)
+ |-- Biology: long (nullable = true)
+ |-- Agriculture: long (nullable = true)
+ |-- Mathematics: long (nullable = true)
+```
+#### 6. Displaying data frame contents 
+It's possible to display the file's contents using the PySpark data frame object's `show()` function.
+```Python
+ourMarks_df.show()
+```
+Output:
+```bash
++--------+---------+-------+-----------+-----------+
+|Division|Kiswahili|Biology|Agriculture|Mathematics|
++--------+---------+-------+-----------+-----------+
+|       C|       80|     70|         85|         91|
+|       B|       80|     70|         85|         91|
+|       A|       80|     78|         96|         92|
+|       A|       92|     70|         89|         96|
++--------+---------+-------+-----------+-----------+
+```
+> To see the whole code for this [click here](https://colab.research.google.com/drive/1Ykwifphdj2mLQ9rhKuX309aXq-tGurTD?usp=sharing)
+### Creating Python data frame from an external file using PySpark
+Our external Files will be imported using the spark session's `read()` function. Data frame object will be created as the result of this action. The techniques to import these file formats are practically comparable, and one may import them with minimal effort. In contrast to the previous method, this one is simple and requires a spark session to produce a PySpark data frame from a Resilient Distributed Dataset.
+
+#### 1. Importing the libraries:
+```Python
+from pyspark.sql import SparkSession
+```
+#### 2. Creating the spark session
+```python
+sparkSession = SparkSession.builder.appName('Creating PySpark dataframe from an external file').getOrCreate()
+```
+Using `appName(),` we can retrieve the application's name by passing in a string. It is possible to leverage an existing spark session or build a new one using the `getOrCreate()` method.
+#### 3. Read the external files into the PySpark data frame
+The external file format that we will use and import comprises a CSV file.
+
+```python
+ourCSV_file = sparkSession.read.csv('IncomeAndGender.csv', sep = ',', inferSchema = True, header = True)
+```
+Delimiters for the CSV file were given (which is a comma). Afterward, the inferSchema property was set to True. CSV files will be automatically transformed into PySpark data frames with this command. 

```

Comment:
Delimiters for the CSV file are given (as a comma). Afterward, the inferSchema property was set to True. CSV files will be automatically transformed into PySpark data frames with this command.